plugins/BEdita/Core/src/Shell/Task/CheckTreeTask.php

```diff
@@ -0,0 +1,284 @@
+                        $entity['uname'],
+                        $entity['id']
+                    )
+                );
+            });
+        } else {
+            $this->verbose('=====> <success>There are no folders that are not in tree.</success>');
+        }
+
+        // Checks on ubiquitous folders.
+        $results = $this->getUbiquitousFolders()
+            ->all();
+        $count = $results->count();
+        if ($count > 0) {
+            $ok = false;
+
+            $this->out(sprintf('=====> <warning>Found %d ubiquitous folders!</warning>', $count));
+            $results->each(function (EntityInterface $entity) {
+                $this->verbose(
+                    sprintf(
+                        '=====>   - folder <info>%s</info> (#<info>%d</info>) is ubiquitous',
+                        $entity['uname'],
+                        $entity['id']
+                    )
+                );
+            });
+        } else {
+            $this->verbose('=====> <success>There are no ubiquitous folders.</success>');
+        }
+
+        // Checks on other objects in root.
+        $results = $this->getObjectsInRoot()
+            ->all();
+        $count = $results->count();
+        if ($count > 0) {
+            $ok = false;
+
+            $this->out(sprintf('=====> <warning>Found %d other objects in root!</warning>', $count));
+            $results->each(function (EntityInterface $entity) {
+                $this->verbose(
+                    sprintf(
+                        '=====>   - %s <info>%s</info> (#<info>%d</info>) is a root',
+                        $this->Objects->ObjectTypes->get($entity['object_type_id'])->get('singular'),
+                        $entity['uname'],
+                        $entity['id']
+                    )
+                );
+            });
+        } else {
+            $this->verbose('=====> <success>There are no other objects in root.</success>');
+        }
+
+        // Checks on other objects with children.
+        $results = $this->getObjectsWithChildren()
+            ->all();
+        $count = $results->count();
+        if ($count > 0) {
+            $ok = false;
+
+            $this->out(sprintf('=====> <warning>Found %d other objects with children!</warning>', $count));
+            $results->each(function (EntityInterface $entity) {
+                $this->verbose(
+                    sprintf(
+                        '=====>   - %s <info>%s</info> (#<info>%d</info>) has children',
+                        $this->Objects->ObjectTypes->get($entity['object_type_id'])->get('singular'),
+                        $entity['uname'],
+                        $entity['id']
+                    )
+                );
+            });
+        } else {
+            $this->verbose('=====> <success>There are no other objects with children.</success>');
+        }
+
+        // Checks on other objects with children.
```

Comment:
typo: I guess comment should be something like `Check objects twice in folder`, or somethink alike

Reply:
Right!